### PR TITLE
sessions_between_states metric

### PR DESCRIPTION
**Pull request by rudigiesler**
Similar to sessions_until_state metric, but allows you to specify the state where you start counting sessions.  This will allow you to evaluate specific sections of an app's performance in terms of session count.

### CODE REVIEW

**Comment by rudigiesler** [comment]
@hodgestar Ready for review

File: lib/metrics-helper.js [hunk 1]
```diff
@@ -267,6 +267,72 @@ var MetricsHelper = Eventable.extend(function(self, im) {
...
+            : self._create_metric_label(
+                'sessions_between', from_action, state_from.state) +
+              self._create_metric_label('', to_action, state_to.state);
+        var metadata_label = 'sessions_between_states_metric_' + label;
+
+        self.im.on('state:' + from_action, function(e) {
+            /* Set counter to zero on state_from */
+            if(e.state.name === state_from.state) {
+                self._set_metadata(
+                    e.state.im.user, metadata_label, 0);
```
Comment (rudigiesler) [inline lib/metrics-helper.js]: I'm not sure if this should be `0` or `1`. `0` makes sense if you think that there were no new sessions between the two states, so the value is `0`, `1` makes sense if you think that in order to get through those states, you had to be in a sessions. I'm open to suggestion here.

Comment (gsvr) [inline lib/metrics-helper.js]: Good question.  I'm leaning towards using 1 here, my thinking is if we use 0 I'd rather call it new_sessions_between_states.  Let's get hodgestar's opinion, though.

Comment (hodgestar) [inline lib/metrics-helper.js]: I'm leaning slightly towards `1` and documenting it.

**Comment by hodgestar** [comment]
Left a reply to your comment. Otherwise looks good!

**Comment by rudigiesler** [comment]
@hodgestar Ready for review again

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**